cdxev/build_public_bom.py

```diff
@@ -1,14 +1,47 @@
 # SPDX-License-Identifier: GPL-3.0-or-later
 
 import json
+import logging
 import re
 import typing as t
 from pathlib import Path
-from typing import Sequence
+from typing import Any, Sequence
 
 from jsonschema import Draft7Validator, FormatChecker
 
 from cdxev.auxiliary.sbomFunctions import extract_components
+from cdxev.log import LogMessage
+
+logger = logging.getLogger(__name__)
+
+
+def check_affected_metadata(metadata: dict[str, Any], path_to_schema: Path) -> bool:
+    """
+    Checks if the metadata.component is affected by the schema.
+    If so, the function prints a warning.
+    Parameters
+    ----------
+    metadata: dict
+        metadata of the SBOM
+    path_to_schema: str
+        str/path to schema
+
+    Returns
+    -------
+    bool
+        True metadata.component is affected,
+        False if not
+    """
+    validator_for_being_internal = create_internal_validator(path_to_schema)
+    if validator_for_being_internal.is_valid(metadata.get("component", [])):
+        logger.warning(
+            LogMessage(
+                "Warning: `metadata.component` is not affected by the JSON schema!",
+                "Please check manually.",
```

Comment:
The warning should be something like "`metadata.component` was not removed even though the JSON schema applies to it. Maybe you try to create an external SBOM for an internal component?"